pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -273,18 +219,9 @@ func newVSphere(cfg VSphereConfig) (*VSphere, error) {
 		cfg.Global.RoundTripperCount = RoundTripperDefaultCount
 	}
 
-	c, err := newClient(context.TODO(), &cfg)
-	if err != nil {
-		return nil, err
-	}
-
-	id, err := getVMName(c, &cfg)
-	if err != nil {
-		return nil, err
-	}
+	id := cfg.Global.VMName
 
 	vs := VSphere{
-		client:          c,
```

Comment:
Nope when you call vSphereLogin, you only renew the client for the first time. It's no difference from previous usage. The only difference is that you don't initialize it at the very beginning but only do that when the master needs to talk to VC.